system.js

```diff
@@ -516,25 +519,32 @@ var gainMoney = function(money, message){
 }
 
 // All pokemon you have gain exp
-var gainExp = function(exp){
+var gainExp = function(exp,level,trainer){
 	if(!isNaN(exp)){
-		exp *= player.expMultiplier
+		console.log(level);
+		console.log(exp);
+		var multiplier = player.expMultiplier;
+		var oakBonus = 1;
 		if(isActive("Exp Share")){
-			exp *= getOakItemBonus("Exp Share")
+			oakBonus = getOakItemBonus("Exp Share");
 		}
 
 		var pokedexBonusExp = pokedexBonus(player.defeatNumbers[curEnemy.id-1]);
-		exp *= pokedexBonusExp;
-
-		exp = Math.floor(exp);
 		var totalMagnitude = getItemBonus("expBoost");
-		exp *= totalMagnitude;
+
+		var trainerBonus = 1;
+		if(trainer == true) {
+			trainerBonus = 1.5;
+		}
+
+
+		var expTotal = Math.floor((exp * trainerBonus * oakBonus * level * multiplier * totalMagnitude) / 7);
```

Comment:
/7 is from the ingame formula: http://bulbapedia.bulbagarden.net/wiki/Experience#Gain_formula